src/Lecture2.hs

```diff
@@ -180,8 +297,11 @@ False
 >>> isIncreasing [1 .. 10]
 True
 -}
+{- Ok, this is rather straightforward. I wonder where's the catch ? -}
 isIncreasing :: [Int] -> Bool
-isIncreasing = error "TODO"
+isIncreasing []         = True
+isIncreasing [x]        = True
+isIncreasing (x : xs)   = x < (head xs) && (isIncreasing xs)
```

Comment:
You can implement `isIncreasing` in a slightly more elegant way by using nested pattern matching 🦥 

```suggestion
isIncreasing (x : y : ys) = x < y && isIncreasing (y : ys)
```